src/player.rs

```diff
@@ -0,0 +1,34 @@
+use ggez::nalgebra::{Vector2, Point2};
+use ggez::{Context, GameResult};
+use ggez::graphics::{Mesh, MeshBuilder, Rect, DrawMode, Color};
+
+const PLAYER_DIMENSIONS: f32 = 30.0;
+
+pub struct Player {
+    location: Vector2<f32>,
+    width: f32,
+    height: f32
+}
+
+impl Player {
+    pub fn new(x: f32, y: f32) -> Player {
+        Player {
+            location: Vector2::new(x, y),
+            width: PLAYER_DIMENSIONS,
+            height: PLAYER_DIMENSIONS
+        }
+    }
+
+    pub fn create_mesh(&self, context: &mut Context, color: Color) -> GameResult<Mesh> {
+        let mesh_bounds: Rect = Rect::new(0.0, 0.0, self.width, self.height);
+        let mesh: Mesh = MeshBuilder::new()
+            .rectangle(DrawMode::fill(), mesh_bounds, color)
+            .build(context)?;
+
+        Ok(mesh)
+    }
+
+    pub fn get_location(&self) -> Point2<f32> {
+        Point2::new(self.location.x, self.location.y)
```

Comment:
Is this going to give the top left corner? If so, that might be an issue later detecting hits.

Reply:
Yes, but you get get player width/height so bounds can be assumed as (x, y), (x+width, y), (x, y+height), (x+width, y+height). 

I'm also not entirely sure how collision will be handled so no point in over optimizing for the future, maybe I'll have some function return a vector of all the bounds or one that accepts a coordinate and returns a boolean of whether or not its in the bounds. 🤷‍♀️